expenses/expensesRouter.js

```diff
@@ -0,0 +1,25 @@
+const express = require('express');
+const router = express.Router();
+
+const Expense = require('./Expense.js');
+
+router.route('/')
+  .get(async (req, res) => {
+    try {
+      const expenses = await Expense.find({}).populate('budget category');
+      res.status(200).json(expenses);
+    } catch (error) {
+      res.status(500).json({ error: "The server has failed you" });
+    }
+  })
+  .post(async (req, res) => {
```

Comment:
We're missing a key element here for the post function, and that's saving refs to the Expense instance.

Here's an example of saving refs to Budget and Category to an Expense:

```
  .post(async (req, res) => {
    const { amount, budget, category, description } = req.body
    try {
      const cat = await Category.find({ title: category })
      const bud = await Budget.find({ title: budget })

      const expense = new Expense({
        amount,
        budget: bud[0]._id,
        category: cat[0]._id,
        description
      })
      await expense.save()
      res.status(200).json(expense)
    } catch (error) {
      res.status(500).json({ error: error.message })
    }
  })
```

Make sure your POST `req.body` has:
```
{
    "description": "description",
    "amount": 210,
    "budget": "aaa",
    "category": "Food/Dining"
}
```
And that `'aaa'` and `'Food/Dining'` exist in Budget and Category DB, respectively

Reply:
Great point, totally forgot that! Honestly, I was not visualizing this project well. Definitely should have stepped back and considered the overall picture better.